File: mobile/types/post.ts
export type Post = {
  id: number;
  user_id: number;
  image_url: string;
  brand: string | null;
  model: string | null;
  latitude: number | null;
  longitude: number | null;
  location_visibility: string;
  is_active: boolean;
  created_at: string;
  expires_at: string;
  user: PostAuthor
  likes_count: number;
  comments_count: number;
  is_liked_by_me: boolean;
};

type PostAuthor = {
  id: number;
  username: string;
  avatar_url: string | null;
};

File: backend/app/models/post.py
from app.models.base import Base
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import String, DateTime, ForeignKey, Boolean, Float
from datetime import datetime, timezone, timedelta

def calculate_expiration() -> datetime:
    return datetime.now(timezone.utc) + timedelta(hours=24)

class Post(Base):
    __tablename__ = "posts"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), index=True, nullable=False)
    image_url: Mapped[str] = mapped_column(String(500), nullable=False)
    brand: Mapped[str | None] = mapped_column(String(255), nullable=True)
    model: Mapped[str | None] = mapped_column(String(255), nullable=True)
    ai_confidence: Mapped[float | None] = mapped_column(Float, nullable=True)
    latitude: Mapped[float | None] = mapped_column(Float, nullable=True)
    longitude: Mapped[float | None] = mapped_column(Float, nullable=True)
    location_visibility: Mapped[str] = mapped_column(String(30), default="public")
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    expires_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=calculate_expiration)
    user: Mapped["User"] = relationship(back_populates="posts")
    likes: Mapped[list["Like"]] = relationship(back_populates="post", passive_deletes=True)
    comments: Mapped[list["Comment"]] = relationship(back_populates="post", passive_deletes=True)
